subsys/bluetooth/audio/shell/bap_broadcast_assistant.c

```diff
@@ -493,39 +472,90 @@ static int cmd_bap_broadcast_assistant_add_src(const struct shell *sh,
 
 static bool broadcast_source_found(struct bt_data *data, void *user_data)
 {
+	static char identified_addr_str[BT_ADDR_LE_STR_LEN];
+	static uint32_t broadcast_id = INVALID_BROADCAST_ID;
 	struct bt_bap_broadcast_assistant_add_src_param param = { 0 };
 	const struct bt_le_scan_recv_info *info = user_data;
 	char addr_str[BT_ADDR_LE_STR_LEN];
 	struct bt_uuid_16 adv_uuid;
-	uint32_t broadcast_id;
+	bool identified_broadcast = false;
 	int err;
 
 	/* Verify that it is a BAP broadcaster*/
 
-	if (data->type != BT_DATA_SVC_DATA16) {
-		return true;
-	}
+	bt_addr_le_to_str(info->addr, addr_str, sizeof(addr_str));
 
-	if (data->data_len < BT_UUID_SIZE_16 + BT_AUDIO_BROADCAST_ID_SIZE) {
-		return true;
-	}
+	switch (data->type) {
+	case BT_DATA_SVC_DATA16:
+		if (data->data_len < BT_UUID_SIZE_16 + BT_AUDIO_BROADCAST_ID_SIZE) {
+			return true;
+		}
 
-	if (!bt_uuid_create(&adv_uuid.uuid, data->data, BT_UUID_SIZE_16)) {
-		return true;
-	}
+		if (!bt_uuid_create(&adv_uuid.uuid, data->data, BT_UUID_SIZE_16)) {
+			return true;
+		}
+
+		if (bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_BROADCAST_AUDIO) != 0) {
+			return true;
+		}
+
+		broadcast_id = sys_get_le24(data->data + BT_UUID_SIZE_16);
+		auto_scan.identified_id = true;
+
+		if (broadcast_id == auto_scan.broadcast_id) {
+			identified_broadcast = true;
+		}
+
+		if (auto_scan.identified_name) {
+			if (!strncmp(identified_addr_str, addr_str, strlen(addr_str))) {
+				identified_broadcast = true;
+			}
+		} else {
+			bt_addr_le_to_str(info->addr, identified_addr_str,
+					sizeof(identified_addr_str));
+		}
+
+		break;
+	case BT_DATA_BROADCAST_NAME:
+		if (strlen(auto_scan.broadcast_name) == 0U) {
+			return true;
+		}
+
+		if (!IN_RANGE(data->data_len, BT_AUDIO_BROADCAST_NAME_LEN_MIN,
+					BT_AUDIO_BROADCAST_NAME_LEN_MAX)) {
+			return true;
+		}
+
+		char broadcast_name[BT_AUDIO_BROADCAST_NAME_LEN_MAX + 1] = { 0 };
+
+		utf8_lcpy(broadcast_name, data->data, data->data_len);
 
-	if (bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_BROADCAST_AUDIO) != 0) {
+		if (!is_substring(auto_scan.broadcast_name, broadcast_name)) {
+			return true;
+		}
+
+		auto_scan.identified_name = true;
+
+		shell_print(ctx_shell, "Found matched broadcast name '%s' with address %s",
+				broadcast_name, addr_str);
+
+		if (auto_scan.identified_id &&
+				!strncmp(identified_addr_str, addr_str, strlen(addr_str))) {
+			identified_broadcast = true;
+		}
```

Comment:
Here we set `identified_broadcast = true`, but we may not have the `broadcast_id` yet

Reply:
There is this condition above it: `if (auto_scan.identified_id && !strncmp(identified_addr_str, addr_str, strlen(addr_str)))`
so `auto_scan.identified_id` flag is true means that we found `broadcast_id` previously, and `!strncmp(identified_addr_str, addr_str, strlen(addr_str))` means that the  `broadcast_id` found previously is at the same addredd as the matched `broadcast_name`

Reply:
Ah, yes, sorry I missed that. The weird indentation makes the code hard to read :D 

But this means that this requires 2 advertising reports in the case that the broadcast name comes first, right?

Reply:
Right. Due to the braodcast_id being static it also works the other way around.